src/palette/internal/paletteprovider.cpp

```diff
@@ -356,7 +356,7 @@ async::Promise<UserPaletteController::RemoveAction> UserPaletteController::showH
         } else if (res.isButton(hideButton)) {
             action = RemoveAction::Hide;
         }
-        resolve(action);
+        return resolve(action);
```

Comment:
@cbjeukendrup ?

Reply:
was already in #30896...

Reply:
No, you created this PR between that one was created and merged ;)

Reply:
Yes, I see that too, now...